### ISSUE TITLE
':PackerCompile' fails

### ISSUE THREAD

**Issue by slbb** [wbthomason/packer.nvim#521]
- `nvim --version`: NVIM v0.5.0-dev+1386-gfd7e77b22
- `git --version`:  2.32.0.windows.1
- Operating system/version: Windows 10 20H2 19042.1110
- Terminal name/version: PowerShell 5

### Steps to reproduce
Update packer.nvim from version older than [this commit](https://github.com/wbthomason/packer.nvim/commit/b76bfba54031ad28f17e98ef555e99cf450d6cb3) to [the latest version so far](https://github.com/wbthomason/packer.nvim/commit/710b9bfc5d557556816329cb2f1317430000fb25).
Then run `:PackerCompile`.

### Actual behaviour
Error occurs.
Here is the error message from packer.log file:
```
[ERROR 7/30/2021 5:13:04 AM 3.8376676207e+14] ...ta\site\pack\packer\opt\packer.nvim\lua\packer\async.lua:20: Error in coroutine: ...vim-data\site\pack\packer\opt\packer.nvim\lua\packer.lua:594: Vim:E492: Not an editor command: -- Automatically generated packer.nvim plugin loader code
```
"packer_compiled.lua" file was created. But error occurs when it's being loaded.

### Expected behaviour
"packer_compiled.lua" file should be loaded successfully.

### The cause of the error
"packer.nvim/packer.lua" line 695 `vim.cmd('source ' .. output_path)` causes the error.
Since "packer_compiled" has been changed from `vim` file to `lua` file in new version of packer.nvim, it should be executed through `luafile`, not `source`.
After modifiying it myself, `:PackerCompiled` works as expected.
However, there's another problem -- neovim won't autoload "$XDG_CONFIG_HOME/nvim/plugin/packer_compiled.lua" on startup, because its not `vim` file. So I have to work around by executing it manually from "init.lua".

Btw, I thiink "packer_compiled.lua" should be placed under "$XDG_CONFIG_HOME/nvim/lua/" folder.

<details>
<summary>Plugin specification file(s)</summary>

Post or link your plugin specification files here, if you aren't able to provide a minimal
reproducer

</details>

<details>
<summary>packer log file</summary>

Post the contents of ~/.cache/nvim/packer.nvim.log here

</details>

<details>
<summary>packer compiled file</summary>

Post the contents of `packer_compiled.vim` here

</details>

**Comment by nanotee** [comment]
> it should be executed through `luafile`, not `source`.

Neovim 0.5 added support for Lua runtime files, you're probably using an outdated nightly build

**Comment by wbthomason** [comment]
Yes, your Neovim needs this PR merged: https://github.com/neovim/neovim/pull/14686 (which, as @nanotee mentions, is included in the 0.5 release).

**Comment by slbb** [comment]
@nanotee @wbthomason Thank you. Solved.
